src/network/networkd-ndisc.c

```diff
@@ -75,6 +77,30 @@ static void ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
                 return;
         }
 
+        SET_FOREACH(address, link->addresses, i) {
+                if (!memcmp(&gateway, &address->in_addr,
+                            sizeof(address->in_addr))) {
```

Comment:
hmmm, so, you are comparing a "struct in6_addr" here with a "union in_addr_union" via memcmp(). Not that it matters much in this case, but I'd prefer if you could compare a "struct in6_addr" with another "struct in6_addr" here, i.e. rewrite this to:

```c
if (memcmp(&gateway, &address->in_addr.in6, sizeof(gateway) == 0) {
…
```

or so would be much prettier, as it clarifies that you actually compare the same stuff here.